check/check.go

```diff
@@ -47,15 +47,30 @@ func handleError(err error, context string) (errmsg string) {
 	return
 }
 
+// Old version - checks don't have sub checks, each check has only one sub check as part of the check itself
+type BaseCheck struct {
+	Audit       string              `json:"audit"`
+	Type        string              `json:"type"`
+	Commands    []*exec.Cmd         `json:"omit"`
+	Tests       *auditeval.Tests    `json:"omit"`
+	Remediation string              `json:"-"`
+	Constraints map[string][]string `yaml:"constraints"`
+}
+
+type SubCheck struct {
+	BaseCheck `yaml:"check"`
+}
+
 // Check contains information about a recommendation.
 type Check struct {
 	ID             string           `yaml:"id" json:"test_number"`
 	Description    string           `json:"test_desc"`
+	Set            bool             `json:"omit"`
+	SubChecks      []SubCheck       `yaml:"sub_checks"`
```

Comment:
I'm pretty sure this will work with an old-format YAML file and a new-format one, but it would be great to have a test case for each to validate that's the case (and to make sure we don't break it in the future)